packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -353,7 +357,60 @@ module.exports = function s3 (config) {
     }, next)
   }
 
+  const policy = {
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Effect: 'Allow',
+        Action: [
+          's3:PutObject',
+        ],
+        Resource: [
+          `arn:aws:s3:::${config.bucket}/*`,
+          `arn:aws:s3:::${config.bucket}`,
+        ],
+      },
+    ],
+  }
+
+  let stsClient
+  function getSTSClient () {
+    if (stsClient == null) {
+      stsClient = new STSClient({
+        region: config.region,
+        credentials : {
+          accessKeyId: config.key,
+          secretAccessKey: config.secret,
+        },
+      })
+    }
+    return stsClient
+  }
+
+  function getTemporarySecurityCredentials (req, res, next) {
+    getSTSClient().send(new GetFederationTokenCommand({
+      Name: '123user',
+      // The duration, in seconds, of the role session. The value specified
+      // can range from 900 seconds (15 minutes) up to the maximum session
+      // duration set for the role.
+      DurationSeconds: config.expires,
+      Policy: JSON.stringify(policy),
+    })).then(response => {
+      res.setHeader('Access-Control-Allow-Origin', '*')
+      res.setHeader('Cache-Control', `public,max-age=${config.expires}`)
```

Comment:
```suggestion
     // this is a public unprotected endpoint.
     // if you implement your own custom endpoint with user authentication you should
     // probably use `private` instead of `public`
      res.setHeader('Cache-Control', `public,max-age=${config.expires}`)
```